toolbox/src/token_2022/wns.rs

```diff
@@ -267,14 +267,15 @@ pub fn approve(accounts: super::wns::ApproveAccounts, params: ApproveParams) ->
     let ending_approve_rent = approve.lamports();
     let ending_payer_lamports = payer.lamports();
 
-    // we take the max value between the minimum rent and the previous rent in case the previous
-    // value is higher than the minimum rent
-    let rent_difference = std::cmp::max(
+    // want to account for potential amount paid in rent.
+    // in case WNS tries to drain to approve account, we cap
+    // the rent difference to the minimum rent.
+    let rent_difference = std::cmp::min(
         Rent::get()?.minimum_balance(APPROVE_LEN),
-        ending_approve_rent,
-    )
-    .checked_sub(initial_approve_rent)
-    .ok_or(TensorError::ArithmeticError)?;
+        ending_approve_rent
+            .checked_sub(initial_approve_rent)
+            .ok_or(TensorError::ArithmeticError)?,
+    );
```

Comment:
This one I'm not sure on: the previous code seems to not really "prevent" much if let's say lamports were siphoned to the approve account maliciously.

Reply:
Yeah I don't think this logic works anyway because both the `approve` and `distribution` account could require additional lamports rent: approve is `init_if_needed` at 16 bytes but the `distribution` account can have arbitrary number of new creators added reallocing to fit so I don't think we can know a priori what an appropriate range is.